### ISSUE TITLE
Error While Reconciling Payment entry

### ISSUE THREAD

**Issue by asoral** [frappe/erpnext#40312]
### Information about bug

```
### Traceback
```
Traceback (most recent call last):
  File "apps/frappe/frappe/app.py", line 110, in application
    response = frappe.api.handle(request)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "apps/frappe/frappe/api/__init__.py", line 49, in handle
    data = endpoint(**arguments)
           ^^^^^^^^^^^^^^^^^^^^^
  File "apps/frappe/frappe/api/v1.py", line 36, in handle_rpc_call
    return frappe.handler.handle()
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "apps/frappe/frappe/handler.py", line 49, in handle
    data = execute_cmd(cmd)
           ^^^^^^^^^^^^^^^^
  File "apps/frappe/frappe/handler.py", line 85, in execute_cmd
    return frappe.call(method, **frappe.form_dict)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "apps/frappe/frappe/__init__.py", line 1716, in call
    return fn(*args, **newargs)
           ^^^^^^^^^^^^^^^^^^^^
  File "apps/frappe/frappe/handler.py", line 327, in run_doc_method
    response = doc.run_method(method)
               ^^^^^^^^^^^^^^^^^^^^^^
  File "apps/frappe/frappe/model/document.py", line 959, in run_method
    out = Document.hook(fn)(self, *args, **kwargs)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "apps/frappe/frappe/model/document.py", line 1319, in composer
    return composed(self, method, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "apps/frappe/frappe/model/document.py", line 1301, in runner
    add_to_return_value(self, fn(self, *args, **kwargs))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "apps/frappe/frappe/model/document.py", line 956, in fn
    return method_object(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "apps/frappe/frappe/utils/typing_validations.py", line 31, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "apps/erpnext/erpnext/accounts/doctype/payment_reconciliation/payment_reconciliation.py", line 536, in reconcile
    self.reconcile_allocations()
  File "apps/erpnext/erpnext/accounts/doctype/payment_reconciliation/payment_reconciliation.py", line 510, in reconcile_allocations
    reconcile_against_document(entry_list, skip_ref_details_update_for_pe, self.dimensions)
  File "apps/erpnext/erpnext/accounts/utils.py", line 511, in reconcile_against_document
    referenced_row = update_reference_in_payment_entry(
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "apps/erpnext/erpnext/accounts/utils.py", line 717, in update_reference_in_payment_entry
    payment_entry.set_missing_ref_details(ref_exchange_rate=d.exchange_rate or None)
TypeError: EmployeePaymentEntry.set_missing_ref_details() got an unexpected keyword argument 'ref_exchange_rate'

```
### Request Data
```
{
	"type": "POST",
	"args": {
		"docs": "{\"name\":\"Payment Reconciliation\",\"minimum_invoice_amount\":0,\"minimum_payment_amount\":0,\"maximum_invoice_amount\":0,\"maximum_payment_amount\":0,\"invoice_limit\":50,\"payment_limit\":50,\"doctype\":\"Payment Reconciliation\",\"invoices\":[{\"docstatus\":0,\"idx\":1,\"invoice_type\":\"Purchase Invoice\",\"invoice_number\":\"PI23-24319-1\",\"invoice_date\":\"2024-03-01\",\"amount\":11762,\"outstanding_amount\":11762,\"currency\":\"INR\",\"exchange_rate\":0,\"parent\":\"Payment Reconciliation\",\"parentfield\":\"invoices\",\"parenttype\":\"Payment Reconciliation\",\"doctype\":\"Payment Reconciliation Invoice\",\"__islocal\":1,\"name\":\"new-payment-reconciliation-invoice-qfizxnriwz\"}],\"payments\":[{\"docstatus\":0,\"idx\":1,\"reference_type\":\"Payment Entry\",\"reference_name\":\"ACC-PAY-2024-00072\",\"posting_date\":\"2024-03-02\",\"amount\":11762,\"difference_amount\":0,\"currency\":\"INR\",\"exchange_rate\":1,\"parent\":\"Payment Reconciliation\",\"parentfield\":\"payments\",\"parenttype\":\"Payment Reconciliation\",\"doctype\":\"Payment Reconciliation Payment\",\"__islocal\":1,\"name\":\"new-payment-reconciliation-payment-odeiqlptfj\"}],\"allocation\":[{\"docstatus\":0,\"idx\":1,\"reference_type\":\"Payment Entry\",\"reference_name\":\"ACC-PAY-2024-00072\",\"invoice_type\":\"Purchase Invoice\",\"invoice_number\":\"PI23-24319-1\",\"allocated_amount\":11762,\"unreconciled_amount\":11762,\"amount\":11762,\"difference_amount\":0,\"difference_account\":\"Exchange Gain/Loss - DTPL\",\"exchange_rate\":1,\"currency\":\"INR\",\"parent\":\"Payment Reconciliation\",\"parentfield\":\"allocation\",\"parenttype\":\"Payment Reconciliation\",\"doctype\":\"Payment Reconciliation Allocation\",\"__islocal\":1,\"name\":\"new-payment-reconciliation-allocation-kbpltuyplz\",\"__checked\":1}],\"party_type\":\"Supplier\",\",\"receivable_payable_account\":\"Creditors - DTPL\",\"__unsaved\":1,\"company\":\"Dexciss Technology Pvt Ltd\",\"docstatus\":0,\"idx\":0}",
		"method": "reconcile"
	},
	"headers": {},
	"error_handlers": {},
	"url": "/api/method/run_doc_method",
	"request_id": "26015f5c-6ffd-4501-991c-d31692385786"
}
```
### Response Data
```
{
	"exception": "TypeError: EmployeePaymentEntry.set_missing_ref_details() got an unexpected keyword argument 'ref_exchange_rate'",
	"exc_type": "TypeError",
	"_exc_source": "erpnext (app)"
}
```

### Module

accounts

### Version

Frappe Version--15.17.0
Erpnext Version--15.16.0

### Installation method

None

### Relevant log output / Stack trace / Full Error Message.

_No response_

**Comment by JeansReal** [comment]
Also the same is using Advance Payments on SI

**Comment by dj12djdjs** [comment]
Please see this issue: https://github.com/frappe/erpnext/issues/40345
Waiting for HRMS to release an update, that repo is causing the bug.

**Comment by YamanQD** [comment]
@dj12djdjs 
We have the same issue, but the version we're using is:

ERPNext: v14.65.0
Frappe Framework: v14.67.0
Frappe HR: v14.0.2
Payments: v0.0.1

It seems that HRMS did release an update, but for v15
How can we fix the issue without upgrading to v15?

**Comment by dj12djdjs** [comment]
@YamanQD HRMS released version-14 update to [v14.22.1](https://github.com/frappe/hrms/releases/tag/v14.22.1) please update Frappe HR to v14.22.1 will solve this.

**Comment by YamanQD** [comment]
> @YamanQD HRMS released version-14 update to [v14.22.1](https://github.com/frappe/hrms/releases/tag/v14.22.1) please update Frappe HR to v14.22.1 will solve this.

I see, looks like there was a problem that got  HRMS stuck at v14.0.2 even after running `bench update`
Fixed it by running `bench get-app hrms --branch version-14`
Thank you!
